common/client-core/src/client/base_client/mod.rs

```diff
@@ -103,6 +104,7 @@ pub struct ClientState {
     pub shared_lane_queue_lengths: LaneQueueLengths,
     pub reply_controller_sender: ReplyControllerSender,
     pub topology_accessor: TopologyAccessor,
+    pub gateway_ws_fd: Option<RawFd>,
```

Comment:
In the `ClientState`, maybe it's useful to wrap this in a slightly higher level `GatewayConnection` struct that could possibly also contain other useful info about the gateway connection state. Idk gateway address, connection state?